### ISSUE TITLE
canonical hostname not resolvable in Rancher 1.2 because /etc/hosts is malformed

### ISSUE THREAD

**Issue by mcanevet** [rancher/rancher#7052]
**Rancher Version:**
1.2.0
**Docker Version:**
1.11.2
**OS and where are the hosts located? (cloud, bare metal, etc):**
Ubuntu Xenial on OpenStack
**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
Single node rancher with external DB
**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**
Cattle
**Steps to Reproduce:**
Create a container and look at the /etc/hosts generated
**Results:** 
```
127.0.0.1	localhost
::1	localhost ip6-localhost ip6-loopback
fe00::0	ip6-localnet
ff00::0	ip6-mcastprefix
ff02::1	ip6-allnodes
ff02::2	ip6-allrouters
	9ec65c353b26
```
**Expected:**
```
127.0.0.1	localhost
::1	localhost ip6-localhost ip6-loopback
fe00::0	ip6-localnet
ff00::0	ip6-mcastprefix
ff02::1	ip6-allnodes
ff02::2	ip6-allrouters
10.42.0.1	9ec65c353b26
```

**Comment by mcanevet** [comment]
Related to #6855

**Comment by mcanevet** [comment]
This is fixed in 1.2.1
